src/unix/fs.c

```diff
@@ -220,6 +220,93 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+#ifdef UV__O_CLOEXEC
+  int (*mkostemp_function)(char*, int);
+  static int no_cloexec_support;
+#endif
+  static const char pattern[] = "XXXXXX";
+  char* path;
+  size_t path_length;
+
+  path = (char*) req->path;
+  path_length = strlen(path);
+
+  /* EINVAL can be returned for 2 reasons:
+       1. The template's last 6 characters were not XXXXXX
+       2. open() didn't support O_CLOEXEC
+     We want to avoid going to the fallback path in case
+     of 1., so it's manually checked before. */
+  if (path_length < strlen(pattern) ||
+      strcmp(path + (path_length - (sizeof(pattern) - 1)), pattern)) {
```

Comment:
here we are computing the length of `pattern` twice, once with strlen (unnecessary) and another one with sizeof - 1. Please either make it a constant or use sizeof on both cases for consistency.